lxd/db/schema/query.go

```diff
@@ -0,0 +1,38 @@
+package schema
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/lxc/lxd/lxd/db/query"
+)
+
+// Return whether the schema table is present in the database.
+func querySchemaTableExists(tx *sql.Tx) (bool, error) {
+	rows, err := tx.Query(stmtSchemaTableExists)
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+	if !rows.Next() {
+		return false, fmt.Errorf("schema table query returned no rows")
+	}
+	var count int
+	if err := rows.Scan(&count); err != nil {
+		return false, err
+	}
+	return count == 1, nil
+}
+
+const stmtSchemaTableExists = `
+SELECT COUNT(name) FROM sqlite_master WHERE type = 'table' AND name = 'schema'
```

Comment:
Why isn't that inside the function? Do you expect that particular statement to be re-used?